src/main/java/org/folio/edge/sip2/MainVerticle.java

```diff
@@ -83,6 +95,20 @@ public void start(Future<Void> startFuture) {
       handlers.put(END_PATRON_SESSION, injector.getInstance(EndPatronSessionHandler.class));
     }
 
+    JsonObject crOptionsJson = config().getJsonObject("tenantConfigRetrieverOptions");
+    ConfigRetrieverOptions crOptions = new ConfigRetrieverOptions(crOptionsJson);
+    ConfigRetriever configRetriever = ConfigRetriever.create(vertx, crOptions);
+
+    configRetriever.getConfig(ar -> {
+      multiTenantConfig = ar.result();
```

Comment:
We should check for errors here. If `ar.failed()`, we should log and possibly try again or just abort.

Also, this is asynchronous, so it is possible the, though highly unlikely, that the socket could start receiving data before the config is retrieved. I'd suggest, we move any code that relies on multiTenantConfig into this lambda or use a Promise to control when the socket starts to listen. A Promise would probably be nicer then embedding the code in this section.

I think something like:
```java
    final Promise<JsonObject> promise = Promise.promise();
    configRetriever.getConfig(promise);
    promise.future().onComplete(ar -> {
      if (ar.failed()) {
        // die?
      } else {
        // get the config from ar.result() and start listening
      }
    });
```

Reply:
good catch, I'll add a Promise